### PR TITLE
implement scrollable message

### PR DESCRIPTION
**Pull request by iCollin**
### Summary
Adds the feature of Scrollable Message

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/js/AppHeader.js [hunk 1]
```diff
@@ -58,6 +60,10 @@ class AppHeader extends React.Component {
         }
     }
 
+    closeScrollable() {
+        uiController.onCloseScrollableMessage(this.props.scrollableMessageMsgId, this.props.scrollableMessageAppId, this.props.activeApp);
```
Comment (JackLivio) [inline src/js/AppHeader.js:64]: Line breaks please :)

File: src/js/AppHeader.js [hunk 2]
```diff
@@ -121,6 +127,15 @@ class AppHeader extends React.Component {
...
                     {alertHtml}
                 </Modal>
+                <Modal
+                isOpen={this.props.showScrollableMessage}
+                className={'app-body scrollableMessageModal'}
+                overlayClassName={`${themeClass} scrollableMessageOverlay`}
+                contentLabel="Example Modal"
+                onRequestClose={this.closeScrollable}
+                >
+                    <ScrollableMessage theme={this.props.theme} body={this.props.scrollableMessageBody} buttons={this.props.softButtons} appName={this.props.scrollableMessageAppName}/>
```
Comment (JackLivio) [inline src/js/AppHeader.js:137]: Line breaks please

File: src/js/Controllers/DisplayCapabilities.js [hunk 3]
```diff
@@ -70,6 +70,7 @@ let textWithGraphicCapabilities = {
 			textField("mainField3"),
 			textField("mainField4"),
 			textField("templateTitle", 50),
+			textField("scrollableMessageBody"),
```
Comment (JackLivio) [inline src/js/Controllers/DisplayCapabilities.js:73]: We might want to include width and row lengths for this capability

File: src/js/Controllers/RpcFactory.js [hunk 4]
```diff
@@ -79,6 +79,29 @@ class RpcFactory {
...
+    }
+    static ScrollableMessageAbortedResponse(rpcID) {
+        return ({
+            "jsonrpc": "2.0",
+            "id": rpcID,
+            "error": {
+                "code": 5,
+                "message": "The Scrollable Message was cancelled",
+                "data": {
+                    "method": "UI.Alert"
```
Comment (JackLivio) [inline src/js/Controllers/RpcFactory.js:100]: Wrong method

File: src/js/Controllers/UIController.js [hunk 5]
```diff
@@ -385,6 +419,15 @@ class UIController {
                    const context = getNextSystemContext();
                    this.onSystemContext(context, rpc.params.appID)
                    return true
+                } else if (rpc.params.functionID === 24 && app.scrollableMessage.active
```
Comment (JackLivio) [inline src/js/Controllers/UIController.js:422]: Scrollable message id is 25

**Comment by JackLivio** [comment]
Try sending a cancel interaction while the scrollable message is displayed when the app is in background/limited. There is a js error.

> Uncaught TypeError: Cannot read property 'scrollableMessage' of undefined

**Comment by iCollin** [comment]
Good catch, CancelInteraction was only working for the `activeApp` before https://github.com/smartdevicelink/generic_hmi/pull/394/commits/f309fc9d52b6c6fadb73edf2fedb32baa925e35c

**Review by JackLivio — APPROVED** [review]

**Pull request merged**